src/shared_utilities/DataHandling.jl

```diff
@@ -0,0 +1,246 @@
+            (target_space, mktempdir(), varname, file_path),
+    )
+    RegridderConstructor = getfield(Regridder, regridder_type)
+    regridder = RegridderConstructor(regridder_args[regridder_type]...)
+
+    # NOTE: this is not concretely typed
+    _cached_regridded_fields = Dict{Dates.DateTime, ClimaCore.Fields.Field}()
+
+    available_dates = file_reader.available_dates
+    times_s = Second.(available_dates .- reference_date) ./ Second(1)
+    available_times = times_s .- t_start
+
+    return DataHandler(
+        file_reader,
+        regridder,
+        target_space,
+        file_reader.dimensions,
+        available_dates,
+        t_start,
+        reference_date,
+        available_times,
+        _cached_regridded_fields,
+    )
+end
+
+"""
+    close(data_handler::DataHandler)
+
+Close any file associated to the given `data_handler`.
+"""
+function Base.close(data_handler::DataHandler)
+    close(data_handler.file_reader)
+    return nothing
+end
+
+"""
+    available_times(data_handler::DataHandler)
+
+Return the time in seconds of the snapshots in the data, measured considering
+the starting time of the simulation and the reference date
+"""
+function available_times(data_handler::DataHandler)
+    return data_handler.available_times
+end
+
+"""
+    available_dates(data_handler::DataHandler)
+
+Return the dates of the snapshots in the data.
+"""
+function available_dates(data_handler::DataHandler)
+    return data_handler.available_dates
+end
+
+"""
+    previous_time(data_handler::DataHandler, time::AbstractFloat)
+    previous_time(data_handler::DataHandler, date::Dates.DateTime)
+
+Return the time in seconds of the snapshot before the given `time`.
+If `time` is one of the snapshots, return itself.
+"""
+function previous_time(data_handler::DataHandler, time::AbstractFloat)
+    time in data_handler.available_times && return time
+    index = searchsortedfirst(data_handler.available_times, time) - 1
+    return data_handler.available_times[index]
+end
+
+function previous_time(data_handler::DataHandler, date::Dates.DateTime)
+    if date in data_handler.available_dates
+        index = searchsortedfirst(data_handler.available_dates, date)
+    else
+        index = searchsortedfirst(data_handler.available_dates, date) - 1
+    end
+    return data_handler.available_times[index]
+end
+
+"""
+    next_time(data_handler::DataHandler, time::AbstractFloat)
+    next_time(data_handler::DataHandler, date::Dates.DateTime)
+
+Return the time in seconds of the snapshot after the given `time`.
+If `time` is one of the snapshots, return the next time.
+"""
+function next_time(data_handler::DataHandler, time::AbstractFloat)
+    index = searchsortedfirst(data_handler.available_times, time)
+    time in data_handler.available_times && (index += 1)
+    return data_handler.available_times[index]
+end
+
+function next_time(data_handler::DataHandler, date::Dates.DateTime)
+    if date in data_handler.available_dates
```

Comment:
maybe worth using the same code  as `next_time` for clarity (otherwise the reader wonders if `next_date` does something different than `next_time`)

Reply:
Addressed in the ClimaUtilities version.